halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/AccountService.java

```diff
@@ -37,36 +38,42 @@
 @Component
 public class AccountService {
   @Autowired
-  ProviderService providerService;
+  LookupService lookupService;
 
   @Autowired
   ValidateService validateService;
 
-  public Account getAccount(NodeReference reference) {
-    Provider provider = providerService.getProvider(reference);
+  public List<Account> getAccounts(NodeReference reference) {
+    NodeFilter filter = new NodeFilter(reference).anyHalconfigFile().anyAccount();
 
-    String accountName = reference.getAccount();
+    return lookupService.getMatchingNodesOfType(filter, Account.class)
+            .stream()
+            .map(n -> (Account) n)
+            .collect(Collectors.toList());
+  }
 
-    List<Account> accounts = provider.getAccounts();
+  public Account getAccount(NodeReference reference) {
```

Comment:
Why does getAccount throw an exception (for not found) but getAccounts does not? It sounds like the caller assumes it exists here but not above.

Though I'm inconsistent, I like to believe that I would use find* to not throw an exception and get* to, at least when I have control over the API naming conventions. So above would be findAllAccounts() and here would be getAccount() or if it were findAccount() it could return null. But youd still throw if there were multiple since it is ambiguous and that isnt expected given the API is assuming a single value.

Had you not had getAccounts() above, I might not thought anything of this as is. The two makes me wonder what the assumed usage pattern is.